Objects/obmalloc.c

```diff
@@ -1233,6 +1243,259 @@ _Py_GetAllocatedBlocks(void)
     return n;
 }
 
+/*==========================================================================*/
+/* radix tree for tracking arena usage
+
+   bit allocation for keys (2^20 arena size)
+
+   64-bit pointers:
+     16 -> ignored (BITS - PHYSICAL_BITS)
+     10 -> MAP_TOP
+     10 -> MAP_MID
+      8 -> MAP_BOT
+     20 -> ideal aligned arena
+   ----
+     64
+
+   32-bit pointers:
+     12 -> MAP_BOT
+     20 -> ideal aligned arena
+   ----
+     32
+
+*/
+
+#if SIZEOF_VOID_P == 8
+
+/* number of bits in a pointer */
+#define BITS 64
+
+/* Current 64-bit processors are limited to 48-bit physical addresses.  For
+ * now, the top 17 bits of addresses will all be equal to bit 2**47.  If that
+ * changes in the future, this must be adjusted upwards.
```

Comment:
Is the 57-bit virtual address space considered?
https://en.wikipedia.org/wiki/Intel_5-level_paging

Reply:
It is only considered in as far as having the PHYSICAL_BITS define.  On systems that implement more address space, we would need some conditional logic to set PHYSICAL_BITS to the appropriate value.  There is a bit of discussion above about it, e.g. PowerPC systems.

Reply:
Thinking about this more, I could add a configure option like `--enable-virtual-addr-size=57` that would allow people to override the PHYSICAL_BITS define.  The calculations for the sizes of the radix tree levels should adapt.

A downside to that approach is if you build for 48-bits and run on a 57-bits addr platform, Python could just crash.   An alternative would be to somehow detect the size of the virtual address space at runtime and then swap out the `address_in_range` function as needed.  I.e. have two implementations of the radix tree and choose the right one.

It seems Go just hard-codes it at 48 bits:
https://github.com/golang/go/blob/6cc8aa7ece96aca282db19f08aa5c98ed13695d9/src/runtime/lfstack_64bit.go#L26
```
	// On s390x, virtual addresses are 64-bit. There's not much we
	// can do about this, so we just hope that the kernel doesn't
	// get to really high addresses and panic if it does.
	addrBits = 48
```

Reply:
OK. I found "To mitigate this, we are not going to allocate virtual address space above 47-bit by default." in the [linux doc](https://www.kernel.org/doc/html/latest/x86/x86_64/5level-paging.html).